internal/api/service/health.go

```diff
@@ -4,10 +4,37 @@ import (
 	"time"
 
 	"github.com/base-org/pessimism/internal/api/models"
+	"github.com/base-org/pessimism/internal/logging"
+	"go.uber.org/zap"
 )
 
 // CheckHealth ... Returns health check for server
-// NOTE - As of now, this is a hardcoded function
 func (svc *PessimismService) CheckHealth() *models.HealthCheck {
-	return &models.HealthCheck{Timestamp: time.Now(), Healthy: true}
+	healthCheck := &models.NodeConnectionStatus{}
+
+	healthCheck.IsL1Healthy = svc.CheckETHRPCHealth(svc.cfg.L1RpcEndpoint)
+	healthCheck.IsL2Healthy = svc.CheckETHRPCHealth(svc.cfg.L2RpcEndpoint)
+
+	healthy := healthCheck.IsL1Healthy && healthCheck.IsL2Healthy
+
+	return &models.HealthCheck{Timestamp: time.Now(), Healthy: healthy}
```

Comment:
We may wanna display/embed the application dependencies that are failing into the response object